integrations-and-supported-tools/pytorch/notebooks/Neptune_PyTorch_Support.ipynb

```diff
@@ -0,0 +1,521 @@
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "90-k5eK205uM"
+      },
+      "source": [
+        "# Basic example"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "gXve6tFt1dLd"
+      },
+      "source": [
+        "**Import libraries**"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "n1NVL5h6MlLq"
+      },
+      "source": [
+        "from numpy.random import permutation\n",
+        "import torch\n",
+        "import torch.nn as nn\n",
+        "import torch.optim as optim\n",
+        "import torchvision\n",
+        "from torchvision import datasets, models, transforms\n",
+        "import matplotlib.pyplot as plt\n",
+        "import neptune.new as neptune\n",
+        "import numpy as np\n",
+        "import seaborn as sns"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "kfqmkNLB1SRu"
+      },
+      "source": [
+        "## Step 1: Create a Neptune run\n",
+        "\n",
+        "To log metadata to the Neptune project, you need the `project name` and the `api_token`.\n",
+        "\n",
+        "To make this example easy to follow, we have created a public project **'common/optuna-integration'** and a shared user **'neptuner'** with the API token **'ANONYMOUS'**. As you will see in the code cell below.\n",
+        "\n",
+        "**(Optional)** To log to your Neptune project:\n",
+        "\n",
+        "* [Create a Neptune account](https://app.neptune.ai/register/)\n",
+        "\n",
+        "* [Find your API token](https://docs.neptune.ai/getting-started/installation#authentication-neptune-api-token)\n",
+        "* [Find your project name](https://docs.neptune.ai/getting-started/installation#setting-the-project-name)\n",
+        "\n",
+        "Pass your credentials to project and api_token arguments of neptune.init()\n",
+        "\n",
+        "`run = neptune.init(api_token='<YOUR_API_TOKEN>', project='<YOUR_WORKSPACE/YOUR_PROJECT>')` # pass your credentials\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "YIAK4NasfQ_f"
+      },
+      "source": [
+        "run = neptune.init(project = 'common/pytorch-integration', tags='Colab Notebook', api_token='ANONYMOUS')"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "a4StPQOmmh_d"
+      },
+      "source": [
+        "Running this cell creates a Run in Neptune, and you can log model building metadata to it.\n",
+        "\n",
+        "**Click on the link above to open the Run in Neptune UI.** For now, it is empty, but you should keep the tab open to see what happens next"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "8CTl-ZOO1Cwr"
+      },
+      "source": [
+        "**Helper functions**"
```

Comment:
I think I'd remove all those helper functions (as well as the fit function) from this notebook.
Also please look at the comments to the examples which can be applied to this notebook as well.

Reply:
Here I took out the fit() and save_image_predictions().

The rest I think it makes sense to leave in place.